src/Tools/dotnet-monitor/Egress/Configuration/IEgressProviderConfigurationProvider.cs

```diff
@@ -4,33 +4,43 @@
 
 using Microsoft.Extensions.Configuration;
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration
 {
     /// <summary>
-    /// Provides access to an Egress:{ProviderType} section of the configuration.
+    /// Provides access to all of the egress provider configuraiton blocks that use the given <see cref="OptionsType"/>.
+    /// These configuration blocks will be from "Egress:{ProviderType}" where "{ProviderType}" is one of the strings definded in <see cref="ProviderTypes"/>.
     /// </summary>
     internal interface IEgressProviderConfigurationProvider
     {
         /// <summary>
-        /// The configuration section associated with the egress provider.
+        /// The name of the categories defined in configuration of this options type. These keys can be passed to <see cref="GetConfigurationSection(string)"/> to get the specific configuration section.
```

Comment:
types instead of categories? We should be consistent in the naming throughout the code and comments.